crates/matrix-sdk-base/src/rooms/normal.rs

```diff
@@ -640,11 +655,41 @@ pub struct RoomInfo {
     pub(crate) members_synced: bool,
     /// The prev batch of this room we received during the last sync.
     pub(crate) last_prev_batch: Option<String>,
+    /// How much we know about this room.
+    #[serde(default = "SyncInfo::complete")] // see fn docs for why we use this default
+    pub(crate) sync_info: SyncInfo,
     /// Base room info which holds some basic event contents important for the
     /// room state.
     pub(crate) base_info: BaseRoomInfo,
 }
 
+#[derive(Clone, Debug, Serialize, Deserialize, PartialEq, Eq)]
+pub(crate) enum SyncInfo {
```

Comment:
You said that you're going to split this out into two sets after I noted that there are 4 states instead of 3 over [here](https://github.com/matrix-org/matrix-rust-sdk/pull/1022#discussion_r971643248). What happened to that idea?

Reply:
It was split, just leaving the 3 states for room-state still (not sure why)